rfc/rfc004-verifiable-transactional-graph.md

```diff
@@ -152,8 +156,54 @@ If the resolved transaction contents is not the latest version, this may indicat
 
 Since the transaction content is detached from the transaction itself and referred to by hash, the transaction content MUST be hashed and compared to the hash specified in the transaction, to assert that the retrieved transaction content is actually the expected transaction content.
 
+### 3.8 Private Transactions
+
+Private transactions are transactions that contain sensitive content, intended for a single specific, or multiple recipients.
+Private transactions MUST be added to the DAG like non-private transactions.
+The recipients MUST be specified as array that contains the node DIDs of the recipients, encrypted separately for each recipient.
+To construct the data to be encrypted, the plaintext recipients MUST be joined by a newline (`\n`)
+
+The recipient list MUST be encrypted for each recipient and the ciphertext specified as array as `pal` header in the JWS.
+
+Given 2 recipients `did:nuts:recipient-A` and `did:nuts:recipient-B`:
+
+```
+encoded_recipients = join("did:nuts:recipient-A", "did:nuts:recipient-B", "\\n")
+
+pal = []
+
+for each recipient
+  encrypted_recipients = ecies_encrypt(encoded_recipients, recipient.encryption_public_key)
+  pal = append(to, encrypted_recipients)
+
+transaction.pal = pal
+```
+
+To decrypt the header, the process above is applied in reverse. If one of the decrypted values match the local node's DID,
+it indicates the local node is (one of) the intended recipient, and it SHOULD try to retrieve the contents.
+
+The encryption key to be used MUST be one of the elliptic curve public encryption keys (`keyAgreement` key in DID documents) of the recipient.
+For encryption of the `pal` header the ECIES encryption algorithm MUST be used.
+
+See appendix A on the reasoning behind encrypting the `pal` header.
+
 ## 4. Example
 
 ```text
 eyJhbGciOiJFUzI1NiIsImN0eSI6ImZvby9iYXIiLCJjcml0IjpbInNpZ3QiLCJ2ZXIiLCJwcmV2cyIsImp3ayJdLCJqd2siOnsia3R5IjoiRUMiLCJjcnYiOiJQLTI1NiIsIngiOiJUTXVzeXNWQTJJcHduNnZFMjhNWUQtOGtPZFN6ajZVTy1MeGE0ZWhLd0d3IiwieSI6IjdZbC1hb2ZPOC1qNHN6aVBYeGREdVVVSXdDSHlaeWtnTTJmdWlISEQxUzgifSwicHJldnMiOlsiMzk3MmRjOTc0NGY2NDk5ZjBmOWIyZGJmNzY2OTZmMmFlN2FkOGFmOWIyM2RkZTY2ZDZhZjg2YzlkZmIzNjk4NiIsImIzZjJjM2MzOTZkYTFhOTQ5ZDIxNGU0YzJmZTBmYzlmYjVmMmE2OGZmMTg2MGRmNGVmMTBjOTgzNWU2MmU3YzEiXSwic2lndCI6MTYwMzQ1Nzk5OSwidmVyIjoxfQ.NDUyZDllODlkNWJkNWQ5MjI1ZmI2ZGFlY2Q1NzllNzM4OGExNjZjNzY2MWNhMDRlNDdmZDNjZDg0NDZlNDYyMA.-jpKBZQ3sc0x34MwnbO8mSiGdUYCfQXNO91RMnvFRq0YZ5pmbKmRYg--zaie-N7wIJhIFbZyuOyJdlcPwZrELQ
 ```
+
+## Appendix A: Correlation Attacks on Private Transactions
```

Comment:
Others RFCs use `## Appendix A: Design decisions` and the A.1 for the first item.